lib/Doctrine/Import/Builder.php

```diff
@@ -865,7 +865,7 @@ public function buildPhpDocs(array $definition)
      * @param string $option
      * @return string assignation code
      */
-    private function emitAssign($level, $name, $option)
+    private function emitAssign($level, $name, $option, &$classname)
```

Comment:
This way to collect the class name is pretty weird, nice trick with a reference.

It is a good opportunity to clean up this method a bit.

I invite you to extract on a new method the "find class matching name".

```suggestion
    private function findTemplateClassMatchingName($name)
    {
        $classname = $name;

        if (class_exists("Doctrine_Template_$name", true)) {
            $classname = "Doctrine_Template_$name";
        }
        
        return $classname;
    }

    private function emitAssign($level, $name, $option, $classname)
    {
        return "        \$" . strtolower($name) . "$level = new $classname($option);". PHP_EOL;
    }
```